sklearn/cluster/tests/test_hdbscan.py

```diff
@@ -28,21 +28,67 @@
 X, y = shuffle(X, y, random_state=7)
 X = StandardScaler().fit_transform(X)
 
-ALGORITHMS = [
+# These are necessary options for both space-tree/MST algorithm selection
+BRUTE_COMPATIBLE = {"auto", "brute"}
+
+ALGORITHMS = {
     "kd_tree",
     "ball_tree",
-    "brute",
-    "auto",
-]
+} | BRUTE_COMPATIBLE
+
+EXACT_MST_ALGORITHMS = {"prims", "boruvka_exact"}
+MST_ALGORITHMS = {"boruvka_approx"} | EXACT_MST_ALGORITHMS | BRUTE_COMPATIBLE
+
```

Comment:
I guess we might want to sorted them once and for all. What do you think?

```suggestion

# Converting sets into sorted list to have a reproducible set suite
# and to prevent errors with test dispatch to pytest-xdist workers.
BRUTE_COMPATIBLE = sorted(BRUTE_COMPATIBLE)
ALGORITHMS = sorted(ALGORITHMS)
EXACT_MST_ALGORITHMS = sorted(EXACT_MST_ALGORITHMS)
MST_ALGORITHMS = sorted(MST_ALGORITHMS)
```